src/events/mqtt.js

```diff
@@ -30,18 +30,18 @@ function publishWrapper(originalPublishFunc) {
                 message,
             };
             const promise = new Promise((resolve) => {
-                patchedCallback = () => {
+                patchedCallback = (err) => {
```

Comment:
this actually seems like a problem i didn't notice before. we should pass all the arguments to the original callback (or we will hurt his application) - you can get `(err, ...rest)` and call `callback(err, ...rest)`

Reply:
if it just takes 2 arguments you can name them as is